src/Avatar.ts

```diff
@@ -112,6 +114,42 @@ export function defaultAvatarUrlForString(s: string): string {
     return dataUrl;
 }
 
+/**
+ * If client is particular about role colors  user role/user directory category then
+ * approach below would be sufficient to replace colors configured in custom config as well as color
+ * You can configure your own color in object in config file
+ * Not having this config or not having color in config parameter would default
+ * avatar colors back to matrix color
+ * configAvatarColors needs to be passed from config and read via config.ts file
+ * configAvatarColors needs to be obj{key:value} pair type in config
+ * @param {string} s
+*/
+function getSpecificColorFromConfig(s: string): string {
+    let configAvatarColors = config.avatarColors;
+    let avatarFromConfigColors = configAvatarColors[s];
+    let numOfColors = Object.keys(configAvatarColors).length;
+    let total = 0;
+    for (let i = 0; i < s.length; ++i) {
+        total += s.charCodeAt(i);
+    }
+    const colorIndex = total % numOfColors;
+    /**
+     * defaultToOtherConfigColors calculates color based on length of username which would remain same even
+     * when page is refreshed and user would like to see same color when page is refreshed.
+     */
+    const defaultToOtherConfigColors = String(Object.values(configAvatarColors)[colorIndex]);
+    /**
+     * randomlySelectedAvatarConfigColor randomly selected color from config
+     * randomly selected color would not be used if there is no 'undefined', 'null' returned
+     * A randomly selected color would be only selected if username is not string
```

Comment:
I don't see how randomlySelectedAvatarConfigColor is ever used.  In the defaultToOtherConfigColors code above, if the for loop is not entered, colorIndex will be 0 and defaultToOtherConfigColors the first configAvatarColors.

Reply:
What do you recommend? I think I should hardcode 7-8 colors there and loop through and show that if config value doesnt match. Alternatively I can remove randomlySelected part. I just did not want matrix color to be applied even when name is not valid string. I think randomlySelectedColor could be used best if some colors are hardcoded so that if config colors are removed accidently then randomly select from hardcoded ones but not matrix ones.

Reply:
I have removed randomly selected color as it was not good experience to see color change on avatars on every page refresh by memory.

Reply:
Update: I have removed randomlyselected color entirely as per my comment above, then later I also refactored further to remove defaultToOtherConfigColor variable so when result is done then at the end just returns finalColor based on given calculations. Also updated feature-communicate.md